File: API/WoodTrack.Core/Models/User/UserModel.cs
namespace WoodTrack.Core.Models;

public class UserModel : BaseModel
{
    public bool IsActive { get; set; }
    public bool IsFirstLogin { get; set; }
    public bool VerificationSent { get; set; }
    public string? FirstName { get; set; } = default!;
    public string? LastName { get; set; } = default!;
    public string UserName { get; set; } = default!;
    public string NormalizedUserName { get; set; } = default!;
    public string Email { get; set; } = default!;
    public string NormalizedEmail { get; set; } = default!;
    public bool EmailConfirmed { get; set; }
    public string PhoneNumber { get; set; } = default!;
    public bool PhoneNumberConfirmed { get; set; }
    public int AccessFailedCount { get; set; }
    public DateTime? BirthDate { get; set; }
    public Gender? Gender { get; set; }
    public string? ProfilePhoto { get; set; }
    public string? ProfilePhotoThumbnail { get; set; }
    public string Address { get; set; } = default!;
    public string? Description { get; set; }
    public string? LicenseNumber { get; set; }
    public int? YearsOfExperience { get; set; }
    public string? WorkingHours { get; set; }
    public string? Position { get; set; }
    public int? CountryId { get; set; }
    public CountryModel? Country { get; set; }
}

File: UI/wood_track_admin/lib/models/user_model.dart
class UserModel {
  bool isActive;
  bool isFirstLogin;
  bool verificationSent;
  String? firstName;
  String? lastName;
  String userName;
  String normalizedUserName;
  String email;
  String normalizedEmail;
  bool emailConfirmed;
  String phoneNumber;
  bool phoneNumberConfirmed;
  int accessFailedCount;
  DateTime? birthDate;
  int? gender;
  String? profilePhoto;
  String? profilePhotoThumbnail;
  String address;
  String? companyName;
  String? description;

  UserModel({
    required this.isActive,
    required this.isFirstLogin,
    required this.verificationSent,
    this.firstName,
    this.lastName,
    required this.userName,
    required this.normalizedUserName,
    required this.email,
    required this.normalizedEmail,
    required this.emailConfirmed,
    required this.phoneNumber,
    required this.phoneNumberConfirmed,
    required this.accessFailedCount,
    this.birthDate,
    this.gender,
    this.profilePhoto,
    this.profilePhotoThumbnail,
    required this.address,
    this.companyName,
    this.description,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      isActive: json['isActive'],
      isFirstLogin: json['isFirstLogin'],
      verificationSent: json['verificationSent'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      userName: json['userName'],
      normalizedUserName: json['normalizedUserName'],
      email: json['email'],
      normalizedEmail: json['normalizedEmail'],
      emailConfirmed: json['emailConfirmed'],
      phoneNumber: json['phoneNumber'],
      phoneNumberConfirmed: json['phoneNumberConfirmed'],
      accessFailedCount: json['accessFailedCount'],
      birthDate: json['birthDate'] != null ? DateTime.parse(json['birthDate']) : null,
      gender: json['gender'],
      profilePhoto: json['profilePhoto'],
      profilePhotoThumbnail: json['profilePhotoThumbnail'],
      address: json['address'],
      companyName: json['companyName'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'isActive': isActive,
      'isFirstLogin': isFirstLogin,
      'verificationSent': verificationSent,
      'firstName': firstName,
      'lastName': lastName,
      'userName': userName,
      'normalizedUserName': normalizedUserName,
      'email': email,
      'normalizedEmail': normalizedEmail,
      'emailConfirmed': emailConfirmed,
      'phoneNumber': phoneNumber,
      'phoneNumberConfirmed': phoneNumberConfirmed,
      'accessFailedCount': accessFailedCount,
      'birthDate': birthDate?.toIso8601String(),
      'gender': gender,
      'profilePhoto': profilePhoto,
      'profilePhotoThumbnail': profilePhotoThumbnail,
      'address': address,
      'companyName': companyName,
      'description': description,
    };
  }
}
